src/main/java/io/cryostat/net/web/http/api/v2/TargetProbePostHandler.java

```diff
@@ -161,13 +166,28 @@ public IntermediateResponse<Void> handle(RequestParameters requestParams) throws
                 getConnectionDescriptorFromParams(requestParams),
                 connection -> {
                     AgentJMXHelper helper = new AgentJMXHelper(connection.getHandle());
-                    helper.defineEventProbes(probeTemplateService.getTemplate(probeTemplate));
+                    String templateContent = probeTemplateService.getTemplate(probeTemplate);
+                    helper.defineEventProbes(templateContent);
+                    List<Event> events = new ArrayList<Event>();
+                    ProbeTemplate template = new ProbeTemplate();
+                    template.deserialize(
+                            new ByteArrayInputStream(
+                                    templateContent.getBytes(StandardCharsets.UTF_8)));
+                    for (Event e : template.getEvents()) {
```

Comment:
I think this could be just:

```java
List<Event> events = new ArrayList<>(template.getEvents());
```

or even

```java
List<Event> events = template.getEvents();
```

since it doesn't look like a writable defensive copy is needed.

Reply:
`template.getEvents` returns type `Event[]` so there is a bit of extra logic needed here. I used `Arrays.asList` instead.